include/termcolor/termcolor.hpp

```diff
@@ -45,17 +52,95 @@
 
 namespace termcolor
 {
-    // Forward declaration of the `__internal` namespace.
-    // All comments are below.
+    //! Since C++ hasn't a way to hide something in the header from
+    //! the outer access, I have to introduce this namespace which
+    //! is used for internal purpose and should't be access from
+    //! the user code.
     namespace __internal
     {
-        inline FILE* get_standard_stream(const std::ostream& stream);
-        inline bool is_atty(const std::ostream& stream);
+        //! Since C++ hasn't a true way to extract stream handler
+        //! from the a given `std::ostream` object, I have to write
+        //! this kind of hack.
+        inline
```

Comment:
Why to move that huge chunk from bottom to above? It seem unrelated to VC++ improvements, isn't it?

Could you please make atomic commits? If you think it should be there, then two separate commits and two pull requests should be sent:

* one, with VC++ improvements
* one, that removes forward declaration

However, I see no benefits with removing forward declaration.